test/cli/fixtures/plugin/vitest.config.ts

```diff
@@ -0,0 +1,41 @@
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  plugins: [
+    {
+      name: "test-default",
+      configureServer() {
+        console.log("##test## configureServer(default)")
```

Comment:
Can we instead inject some kind of global and populate it instead of logging? I hate debugging these tests usually :S

Reply:
Logging was actually useful for debugging since I can directly see it when running a fixture:

```sh
$ pnpm -C test/cli test -- --root fixtures/plugin

> @vitest/test-cli@ test /home/hiroshi/code/others/vitest/test/cli
> vitest "--root" "fixtures/plugin"

##test## configureServer(pre)
##test## configureServer(default)
##test## buildStart(pre)
##test## buildStart(default)

 DEV  v1.5.3 /home/hiroshi/code/others/vitest/test/cli/fixtures/plugin

##test## resolveId(pre) vitest
##test## resolveId(pre) vitest
##test## resolveId(pre) /@fs/home/hiroshi/code/others/vitest/node_modules/.pnpm/vite@5.2.6_@types+node@20.11.5/node_modules/vite/dist/client/env.mjs
##test## resolveId(pre) /@fs/home/hiroshi/code/others/vitest/node_modules/.pnpm/vite@5.2.6_@types+node@20.11.5/node_modules/vite/dist/client/env.mjs
##test## transform(pre) /home/hiroshi/code/others/vitest/node_modules/.pnpm/vite@5.2.6_@types+node@20.11.5/node_modules/vite/dist/client/env.mjs
##test## transform(default) /home/hiroshi/code/others/vitest/node_modules/.pnpm/vite@5.2.6_@types+node@20.11.5/node_modules/vite/dist/client/env.mjs
##test## resolveId(pre) /home/hiroshi/code/others/vitest/packages/vitest/dist/spy.js
##test## resolveId(pre) /home/hiroshi/code/others/vitest/test/cli/fixtures/plugin/basic.test.ts
##test## resolveId(pre) /home/hiroshi/code/others/vitest/test/cli/fixtures/plugin/basic.test.ts
##test## transform(pre) /home/hiroshi/code/others/vitest/test/cli/fixtures/plugin/basic.test.ts
##test## transform(default) /home/hiroshi/code/others/vitest/test/cli/fixtures/plugin/basic.test.ts
##test## resolveId(pre) vitest
##test## resolveId(pre) vitest
##test## resolveId(pre) /@fs/home/hiroshi/code/others/vitest/packages/vitest/dist/index.js
 ✓ basic.test.ts (1)
   ✓ basic
```

It might not good for writing assertions, so I'll replace with globalThis, but I kinda want to keep the log still.

Reply:
Let's remove the log, there is a reason why we removed log interceptions from other tests